state/remoteentities.go

```diff
@@ -262,6 +262,15 @@ func (r *RemoteEntities) SaveMacaroon(entity names.Tag, mac *macaroon.Macaroon)
 		macJSON = string(b)
 	}
 	buildTxn := func(attempt int) ([]txn.Op, error) {
+		if attempt > 0 {
+			// The entity may have been removed; if so,
+			// return a not found error.
```

Comment:
what?

Reply:
we want to return a not found error rather than a "state changing too quickly" error